src/main/java/xyz/nucleoid/plasmid/game/common/team/TeamManager.java

```diff
@@ -0,0 +1,330 @@
+package xyz.nucleoid.plasmid.game.common.team;
+
+import it.unimi.dsi.fastutil.objects.Object2BooleanArrayMap;
+import it.unimi.dsi.fastutil.objects.Object2BooleanMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.Packet;
+import net.minecraft.network.packet.s2c.play.TeamS2CPacket;
+import net.minecraft.scoreboard.AbstractTeam;
+import net.minecraft.scoreboard.Scoreboard;
+import net.minecraft.scoreboard.Team;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Style;
+import net.minecraft.text.Text;
+import net.minecraft.util.ActionResult;
+import org.jetbrains.annotations.Nullable;
+import xyz.nucleoid.plasmid.game.GameActivity;
+import xyz.nucleoid.plasmid.game.event.GamePlayerEvents;
+import xyz.nucleoid.plasmid.game.player.MutablePlayerSet;
+import xyz.nucleoid.plasmid.game.player.PlayerSet;
+import xyz.nucleoid.plasmid.util.PlayerRef;
+import xyz.nucleoid.stimuli.event.player.PlayerDamageEvent;
+
+import java.util.*;
+
+/**
+ * Simple, GameActivity specific team manager class
+ */
+public final class TeamManager {
+    private final Map<GameTeam, Set<PlayerRef>> teamToPlayerRefSet = new Object2ObjectOpenHashMap<>();
```

Comment:
I'm a bit confused by the difference between `teamToPlayerRefSet` and `teamToPlayerSet` here- do they not serve the same purpose? They don't seem to be used consistently (e.g. in the player add listener)

Reply:
`teamToPlayerRefSet` contains set of all players that are added (including offline ones and ones that left the minigame).
`teamToPlayerSet` contains only players that are in the game

Reply:
`PlayerSet` is interesting since it can contain both online and offline players, but the standard iterator only yields online. It might be possible to merge them and still be useful. Given https://github.com/NucleoidMC/plasmid/commit/4b4815e77929a82d0fba2be03847ee9502bdd3c2, it should be possible to access everything needed through the `PlayerSet`.

Reply:
A further note- given there's quite a few `Map<GameTeam, T>` it might be useful to combine them into a single state object.

Reply:
I will merge these maps. But with merging refSet and PlayerSet, I'm not sure. I prefer to keep PlayerSet as it's own thing, so it's method won't apply to players that left, while still keeping reference to them